### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by ragini07**
Authentication 
- private Routes ( cart , profile and user ) are redirected to login page if user not logged in
- login with admin credential or with created user ( can try using this cred  email : test@gmail.com password : Test@123 )
- on Logging in user can logout from profile page
- User data displayed on profile page
- sign up new user and log out from profile page

-Navigate to products page from Home page by selecting Featured Category 

Note : form validation not added yet , password show hide not added yet

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -9,6 +9,7 @@
       name="description"
       content="Web site created using create-react-app"
     />
+    <!-- <link rel="stylesheet" href="https://unpkg.com/mockman-js@latest/dist/style.css" /> -->
```
Comment (Vikas2426) [inline public/index.html]: should not commit commented code

File: src/App.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import "./App.css";
-import { Header ,Login ,SignUp , Products , Cart,WishList, NotFound, Home} from "./Components/index.js";
+import { UserProfile,Header ,Login ,SignUp , Products , Cart,WishList, NotFound, Home , ProductDetail} from "./Components/index.js";
```
Comment (Vikas2426) [inline src/App.js]: no need to write index.js  "./Components" will work the same

File: src/Components/Auth/SignUp.js [hunk 3]
```diff
@@ -1,8 +1,59 @@
...
+                    value={formData.email}
+                    onChange={(e) => setFormData({...formData , email : e.target.value})}/></div>
+           <div className="input-with-eye">
+                  <input 
+                      type="password" 
+                      placeholder="password"
+                      value={formData.password}
+                      onChange={(e) => setFormData({...formData , password : e.target.value})}/>
+            <i className="fa fa-eye btn-icon eye-icon"></i></div>
+           {/* <div className="input-with-eye"><input type="password" placeholder="confirm password"/>
```
Comment (Vikas2426) [inline src/Components/Auth/SignUp.js:47]: commented code

File: src/Components/ProductDetail/ProductDetail.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import {Loader} from '../index'
+import { useProducts } from '../../Context/products-context'
+
+function ProductDetail() {
+    const [product , setProduct] = useState({})
+    const [isLoading, setIsLoading] = useState(false)
+
+    const {id} = useParams();
+    console.log(id)
+    // const {products} = useProducts()
```
Comment (Vikas2426) [inline src/Components/ProductDetail/ProductDetail.js]: remove commented code

File: src/Components/ProductDetail/ProductDetail.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import axios from 'axios';
+import {Loader} from '../index'
+import { useProducts } from '../../Context/products-context'
+
+function ProductDetail() {
+    const [product , setProduct] = useState({})
+    const [isLoading, setIsLoading] = useState(false)
+
+    const {id} = useParams();
+    console.log(id)
```
Comment (Vikas2426) [inline src/Components/ProductDetail/ProductDetail.js]: remove console.log

File: src/Context/auth-context.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+import { authReducerFtn } from '../Reducer/authReducer';
+
+
+const authContext = createContext();
+
+export const AuthProvider = ({children}) => {
+
+    
+
+
```
Comment (Vikas2426) [inline src/Context/auth-context.js]: remove empty lines

File: src/Context/auth-context.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    
+
+ 
+    const authToken = JSON.parse(localStorage.getItem('authToken'))
+    const userData = JSON.parse(localStorage.getItem('userData'))
+    const [token , setToken] = useState(authToken?.token)
+    const [user , setUser] = useState(userData?.user)
+
+
+
```
Comment (Vikas2426) [inline src/Context/auth-context.js]: empty lines

File: src/backend/controllers/AuthController.js [hunk 8]
```diff
@@ -66,9 +66,12 @@ export const signupHandler = function (schema, request) {
 
 export const loginHandler = function (schema, request) {
   const { email, password } = JSON.parse(request.requestBody);
+  console.log("backend"+email , password)
   try {
     const foundUser = schema.users.findBy({ email });
+    console.log("found user"+foundUser)
     if (!foundUser) {
+      console.log()
```
Comment (Vikas2426) [inline src/backend/controllers/AuthController.js]: remove console.log

File: src/backend/controllers/AuthController.js [hunk 9]
```diff
@@ -93,6 +96,7 @@ export const loginHandler = function (schema, request) {
       }
     );
   } catch (error) {
+    console.log({error})
```
Comment (Vikas2426) [inline src/backend/controllers/AuthController.js:99]: use a toast to show the error message, dont console.log

**Review by Vikas2426 — APPROVED** [review]
left a few important suggestions

File: src/Reducer/authReducer.js [hunk 10]
```diff
@@ -0,0 +1,3 @@
+export const authReducerFtn = (state,action) => {
+    
+}
```
Comment (manismk) [inline src/Reducer/authReducer.js]: If this file not needed you can remove this

**Review by manismk — APPROVED** [review]
Remove the console logs as vikas suggested.

Everything looks great. Detailed naming convention and easy to understand

File: public/index.html (hunk 1 above)
Comment (ragini07) [reply public/index.html]: removed it

File: src/App.js (hunk 2 above)
Comment (ragini07) [reply src/App.js]: Thanks for noticing.. done the change

File: src/Components/ProductDetail/ProductDetail.js (hunk 4 above)
Comment (ragini07) [reply src/Components/ProductDetail/ProductDetail.js]: done removed all console log and commented code

File: src/Context/auth-context.js (hunk 6 above)
Comment (ragini07) [reply src/Context/auth-context.js]: done

File: src/Components/Auth/SignUp.js (hunk 3 above)
Comment (ragini07) [reply src/Components/Auth/SignUp.js:47]: I will use this piece of code in future so commented for now

File: src/Components/ProductDetail/ProductDetail.js (hunk 5 above)
Comment (ragini07) [reply src/Components/ProductDetail/ProductDetail.js]: done

File: src/backend/controllers/AuthController.js (hunk 8 above)
Comment (ragini07) [reply src/backend/controllers/AuthController.js]: done

File: src/backend/controllers/AuthController.js (hunk 9 above)
Comment (ragini07) [reply src/backend/controllers/AuthController.js:99]: yes Thankyou again .Will add toastify later

File: src/Reducer/authReducer.js (hunk 10 above)
Comment (ragini07) [reply src/Reducer/authReducer.js]: done

Comment (ragini07) [reply src/Reducer/authReducer.js]: done

File: src/Context/auth-context.js (hunk 7 above)
Comment (ragini07) [reply src/Context/auth-context.js]: removed empty space

**Pull request merged**